Please support changing timeout.

https://github.com/influxdata/influxdb-client-php/blob/4de202d09a7c82796d45c3ebffc0cbdb37b470f7/src/InfluxDB2/DefaultApi.php#L13

The default timeouts are massive, even when i turn all retry's off the 10s is still too much.
Could it be possible to change it in options?

`$this->options['timeout'] ?? self::DEFAULT_TIMEOUT`

@bednar The PR is now completed. But the make test function does not work. I can debug that later, if it's needed.

```
Traceback (most recent call last):
  File "urllib3/connectionpool.py", line 677, in urlopen
  File "urllib3/connectionpool.py", line 392, in _make_request
  File "http/client.py", line 1277, in request
  File "http/client.py", line 1323, in _send_request
  File "http/client.py", line 1272, in endheaders
  File "http/client.py", line 1032, in _send_output
  File "http/client.py", line 972, in send
  File "docker/transport/unixconn.py", line 43, in connect
FileNotFoundError: [Errno 2] No such file or directory

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "requests/adapters.py", line 449, in send
  File "urllib3/connectionpool.py", line 727, in urlopen
  File "urllib3/util/retry.py", line 410, in increment
  File "urllib3/packages/six.py", line 734, in reraise
  File "urllib3/connectionpool.py", line 677, in urlopen
  File "urllib3/connectionpool.py", line 392, in _make_request
  File "http/client.py", line 1277, in request
  File "http/client.py", line 1323, in _send_request
  File "http/client.py", line 1272, in endheaders
  File "http/client.py", line 1032, in _send_output
  File "http/client.py", line 972, in send
  File "docker/transport/unixconn.py", line 43, in connect
urllib3.exceptions.ProtocolError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker/api/client.py", line 214, in _retrieve_server_version
  File "docker/api/daemon.py", line 181, in version
  File "docker/utils/decorators.py", line 46, in inner
  File "docker/api/client.py", line 237, in _get
  File "requests/sessions.py", line 543, in get
  File "requests/sessions.py", line 530, in request
  File "requests/sessions.py", line 643, in send
  File "requests/adapters.py", line 498, in send
requests.exceptions.ConnectionError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 81, in main
  File "compose/cli/main.py", line 198, in perform_command
  File "compose/cli/command.py", line 70, in project_from_options
  File "compose/cli/command.py", line 153, in get_project
  File "compose/cli/docker_client.py", line 43, in get_client
  File "compose/cli/docker_client.py", line 170, in docker_client
  File "docker/api/client.py", line 197, in __init__
  File "docker/api/client.py", line 222, in _retrieve_server_version
docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
[9194] Failed to execute script docker-compose
Makefile:23: recipe for target 'start-server' failed
make: *** [start-server] Error 255
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,10 +1,10 @@
 ## 1.12.0 [unreleased]
 
 ### Features
-1. [#68](https://github.com/influxdata/influxdb-client-php/pull/68): Moved classes to their own file
+1. [#74](https://github.com/influxdata/influxdb-client-php/pull/74): Allow to change GuzzleHTTP timeout
 
 ### API
-1. [#71](https://github.com/influxdata/influxdb-client-php/pull/71): Updated swagger to the latest version
+1. [#71](https://github.com/influxdata/influxdb-client-php/pull/71): Update swagger to the latest version
 
 ## 1.11.0 [2021-03-05]
 

--- a/README.md
+++ b/README.md
@@ -65,7 +65,7 @@ $client = new InfluxDB2\Client([
 | org | Default organization bucket for writes | String | none |
 | precision | Default precision for the unix timestamps within the body line-protocol | String | none |
 | verifySSL | Turn on/off SSL certificate verification. Set to `false` to disable certificate verification. | bool | true |
-
+| timeout | Describing the number of seconds to wait while trying to connect to a server. Use 0 to wait indefinitely | int | 10 |
 
 
 ### Queries

--- a/src/InfluxDB2/Client.php
+++ b/src/InfluxDB2/Client.php
@@ -32,7 +32,8 @@ class Client
      *          "debug" => false,
      *          "logFile" => "php://output",
      *          "tags" => ['id' => '1234',
-     *              'hostname' => '${env.Hostname}']
+     *              'hostname' => '${env.Hostname}'],
+     *          "timeout" => 2
      *          ]);
      *
      * @param array $options

--- a/src/InfluxDB2/DefaultApi.php
+++ b/src/InfluxDB2/DefaultApi.php
@@ -14,7 +14,12 @@ class DefaultApi
     public $options;
     /** @var Client */
     public $http;
-
+    /**
+     * Holds GuzzleHttp timeout.
+     *
+     * @var int
+     */
+    private $timeout;
     /**
      * DefaultApi constructor.
      * @param $options
@@ -22,10 +27,10 @@ class DefaultApi
     public function __construct(array $options)
     {
         $this->options = $options;
-
+        $this->timeout = $this->options['timeout'] ?? self::DEFAULT_TIMEOUT;
         $this->http = new Client([
             'base_uri' => $this->options['url'],
-            'timeout' => self::DEFAULT_TIMEOUT,
+            'timeout' => $this->timeout,
             'verify' => $this->options['verifySSL'] ?? true,
             'headers' => [
                 'Authorization' => "Token {$this->options['token']}"
@@ -41,17 +46,17 @@ class DefaultApi
      * @param bool $stream - use streaming
      * @return ResponseInterface
      */
-    public function post($payload, $uriPath, $queryParams, $timeout = self::DEFAULT_TIMEOUT, bool $stream = false): ResponseInterface
+    public function post($payload, $uriPath, $queryParams, $timeout = null, bool $stream = false): ResponseInterface
     {
         return $this->request($payload, $uriPath, $queryParams, 'POST', $timeout, $stream);
     }
 
-    public function get($payload, $uriPath, $queryParams, $timeout = self::DEFAULT_TIMEOUT): ResponseInterface
+    public function get($payload, $uriPath, $queryParams, $timeout = null): ResponseInterface
     {
         return $this->request($payload, $uriPath, $queryParams, 'GET', $timeout, false);
     }
 
-    private function request($payload, $uriPath, $queryParams, $method, $timeout = self::DEFAULT_TIMEOUT, bool $stream = false): ResponseInterface
+    private function request($payload, $uriPath, $queryParams, $method, $timeout = null, bool $stream = false): ResponseInterface
     {
         try {
             $options = [
@@ -63,7 +68,7 @@ class DefaultApi
                 'query' => $queryParams,
                 'body' => $payload,
                 'stream' => $stream,
-                'timeout' => $timeout
+                'timeout' => $timeout ?? $this->timeout
             ];
 
             // enable debug

